### DISCUSSION TITLE
My WfH Control Pannel Stream Deck

### DISCUSSION THREAD

**Discussion by CommanderB** [ybizeul/StreamDeckWS#16] [Show and tell]
This is my work from home 'control center' implemented on a StreamDeck. Pre StreamDeckWS this was a big kludge, now all very elegant with websockets to Node-Red flows

![SD 1](https://user-images.githubusercontent.com/18680188/104954584-7d794c80-5996-11eb-9623-e3af79a922d8.png)

The three left buttons control a Raspberry Pi Zero based 'traffic light' in the main part of the house which subscribes to an MQTT topic and displays my availability for the rest of the family.

The next column controls use Toggl via `node-red-contrib-toggl` to track my sit / stand time during my working day. 

The audio controls are for the Sonos in my room via `node-red-contrib-better-sonos`. Also linked to the traffic lights to play on green, pause on red and amber.

The blinds are Ikea Fyrtur controlled through `node-red-contrib-ikea-home-smart` and the ZWave lights are controlled via Indigo for Mac using `node-red-contrib-indigo`

![SD 2](https://user-images.githubusercontent.com/18680188/104955165-c1b91c80-5997-11eb-8f74-525b0f226afd.png)

All icon statuses update on StreamDeck even if controlled from elsewhere (lights from switch etc)

![SD 3](https://user-images.githubusercontent.com/18680188/104955263-ed3c0700-5997-11eb-9031-6f4f03bb860d.png)

**Comment by ybizeul** [comment]
Excellent use case ! Thanks for the writeup, I hope it's going to inspire others !

**Reply by CommanderB** [reply]
I can't tell you how good this is! I keep tweaking my SD. The latest is the clock which has been implemented in node-red with an updating background colour at the top of the hour as a reminder of meeting starts.

![Untitled](https://user-images.githubusercontent.com/18680188/105765411-67373780-5f26-11eb-8a04-dab079dc7b51.png)

I also filled in the unused key to report temperature pulled from the thermostat

**Comment by CommanderB** [comment]
More tweaks - Now the album art from the local Sonos is shown when playing using events from `node-red-contrib-sonos-events` and reverts to a play key when nothing is playing. 

![Untitled 3](https://user-images.githubusercontent.com/18680188/106388917-962e2d00-63ae-11eb-9a39-40860ec535a2.png)

**Comment by CommanderB** [comment]
And then I had some fun manipulating the images using `node-red-contrib-image-tools` Not right for my setup, but proof of what is possible

![Untitled 2](https://user-images.githubusercontent.com/18680188/106394542-13679b00-63cb-11eb-80fd-afba009cbb65.png)
![Untitled 4](https://user-images.githubusercontent.com/18680188/106394546-1793b880-63cb-11eb-9a4c-905f40a12124.png)
